### ISSUE TITLE
[Paper Note] WizardLM: Empowering large pre-trained language models to follow complex instructions, Can Xu+, arXiv'23, 2023.04

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#533]
# URL
- https://arxiv.org/abs/2304.12244
# Authors
  - Can Xu
  - Qingfeng Sun
  - Kai Zheng
  - Xiubo Geng
  - Pu Zhao
  - Jiazhan Feng
  - Chongyang Tao
  - Qingwei Lin
  - Daxin Jiang
# Abstract
  - Training large language models (LLMs) with open-domain instruction following data brings colossal success. However, manually creating such instruction data is very time-consuming and labor-intensive. Moreover, humans may struggle to produce high-complexity instructions. In this paper, we show an avenue for creating large amounts of instruction data with varying levels of complexity using LLM instead of humans. Starting with an initial set of instructions, we use our proposed Evol-Instruct to rewrite them step by step into more complex instructions. Then, we mix all generated instruction data to fine-tune LLaMA. We call the resulting model WizardLM. Human evaluations on a complexity-balanced test bed and Vicuna's testset show that instructions from Evol-Instruct are superior to human-created ones. By analyzing the human evaluation results of the high complexity part, we demonstrate that outputs from our WizardLM are preferred to outputs from OpenAI ChatGPT. In GPT-4 automatic evaluation, WizardLM achieves more than 90\% capacity of ChatGPT on 17 out of 29 skills. Even though WizardLM still lags behind ChatGPT in some aspects, our findings suggest that fine-tuning with AI-evolved instructions is a promising direction for enhancing LLMs. Our code and data are public at https://github.com/nlpxucan/WizardLM
# Translation (by gpt-4o-mini)
- 大規模言語モデル（LLMs）をオープンドメインの指示に従うデータで訓練することは、巨大な成功をもたらしています。しかし、そのような指示データを手動で作成することは非常に時間がかかり、労力を要します。さらに、人間は高い複雑性を持つ指示を生成するのに苦労することがあります。本論文では、人間の代わりにLLMを使用して、さまざまな複雑さのレベルを持つ大量の指示データを作成する方法を示します。初期の指示セットから始めて、提案するEvol-Instructを使用して、それらを段階的により複雑な指示に書き換えます。その後、生成されたすべての指示データを混ぜてLLaMAをファインチューニングします。これにより得られたモデルをWizardLMと呼びます。複雑さがバランスされたテストベッドとVicunaのテストセットにおける人間の評価は、Evol-Instructからの指示が人間が作成したものよりも優れていることを示しています。高い複雑性の部分における人間の評価結果を分析することで、WizardLMからの出力がOpenAI ChatGPTからの出力よりも好まれることを示します。GPT-4の自動評価では、WizardLMは29のスキルのうち17でChatGPTの90%以上の能力を達成しています。WizardLMはまだいくつかの側面でChatGPTに劣っていますが、AI進化による指示でのファインチューニングがLLMを強化するための有望な方向性であることを示唆しています。私たちのコードとデータは、https://github.com/nlpxucan/WizardLM で公開されています。
# Summary (by gpt-4o-mini)
- 本論文では、LLMを用いて複雑な指示データを自動生成する方法を提案。Evol-Instructを使用して初期の指示を段階的に書き換え、生成したデータでLLaMAをファインチューニングし、WizardLMモデルを構築。評価結果は、Evol-Instructからの指示が人間作成のものより優れており、WizardLMがChatGPTよりも高い評価を得ることを示す。AI進化による指示生成がLLM強化の有望なアプローチであることを示唆。

**Comment by AkihikoWatanabe** [comment]
instruction trainingは大きな成功を収めているが、人間がそれらのデータを作成するのはコストがかかる。また、そもそも複雑なinstructionを人間が作成するのは苦労する。そこで、LLMに自動的に作成させる手法を提案している（これはself instructと一緒）。データを生成する際は、seed setから始め、step by stepでinstructionをrewriteし、より複雑なinstructionとなるようにしていく。
これらの多段的な複雑度を持つinstructionをLLaMaベースのモデルに食わせてfinetuningした（これをWizardLMと呼ぶ）。人手評価の結果、WizardLMがChatGPTよりも好ましいレスポンスをすることを示した。特に、WizaraLMはコード生成や、数値計算といった難しいタスクで改善を示しており、複雑なinstructionを学習に利用することの重要性を示唆している。

**Comment by AkihikoWatanabe** [comment]
EvolInstructを提案。"1+1=?"といったシンプルなinstructionからスタートし、これをLLMを利用して段階的にcomplexにしていく。complexにする方法は2通り：
- In-Depth Evolving: instructionを5種類のoperationで深掘りする（blue direction line）
  - add constraints
  - deepening
  - concretizing
  - increase reasoning steps
  - complicate input
- In-breadth Evolving: givenなinstructionから新しいinstructionを生成する

上記のEvolvingは特定のpromptを与えることで実行される。
また、LLMはEvolvingに失敗することがあるので、Elimination Evolvingと呼ばれるフィルタを利用してスクリーニングした。
フィルタリングでは4種類の失敗するsituationを想定し、1つではLLMを利用。2枚目画像のようなinstructionでフィルタリング。
1. instructionの情報量が増えていない場合。
2. instructionがLLMによって応答困難な場合（短すぎる場合やsorryと言っている場合）
3. puctuationやstop wordsによってのみ構成されている場合 
4.明らかにpromptの中から単語をコピーしただけのinstruction（given prompt, rewritten prompt, #Rewritten Prompt#など）
![image](https://user-images.githubusercontent.com/12249301/234436445-e84ff44e-7b0b-4217-a735-7444b04bd760.png)
![image](https://user-images.githubusercontent.com/12249301/234437210-6cb6d75f-509a-4f2e-a767-dba8861d8a69.png)

**Comment by AkihikoWatanabe** [comment]
__translate: Training large language models (LLMs) with open-domain instruction following
data brings colossal success. However, manually creating such instruction data
is very time-consuming and labor-intensive. Moreover, humans may struggle to
produce high-complexity instructions. In this paper, we show an avenue for creating
large amounts of instruction data with varying levels of complexity using LLM
instead of humans. Starting with an initial set of instructions, we use our proposed
Evol-Instruct to rewrite them step by step into more complex instructions. Then, we
mix all generated instruction data to fine-tune LLaMA. We call the resulting model
WizardLM. Human evaluations on a complexity-balanced test bed and Vicuna’s
testset show that instructions from Evol-Instruct are superior to human-created
ones. By analyzing the human evaluation results of the high complexity part, we
demonstrate that outputs from our WizardLM model are preferred to outputs from
OpenAI ChatGPT. In GPT-4 automatic evaluation, WizardLM achieves more than
90% capacity of ChatGPT on 17 out of 29 skills. Even though WizardLM still
lags behind ChatGPT in some aspects, our findings suggest that fine-tuning with
AI-evolved instructions is a promising direction for enhancing LLMs. Our code
and data are public at https://github.com/nlpxucan/WizardLM.
